flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PartitionedFileReader.java

```diff
@@ -61,104 +56,69 @@
+    private long nextOffsetToRead;
+
     /** Number of remaining buffers in the current data region read. */
     private int currentRegionRemainingBuffers;
 
-    /** Whether this partitioned file reader is closed. */
-    private boolean isClosed;
-
-    public PartitionedFileReader(PartitionedFile partitionedFile, int targetSubpartition)
+    public PartitionedFileReader(
+            PartitionedFile partitionedFile,
+            int targetSubpartition,
+            FileChannel dataFileChannel,
+            FileChannel indexFileChannel)
             throws IOException {
+        checkArgument(
+                dataFileChannel.isOpen() && indexFileChannel.isOpen(),
+                "Both data file channel and index file channel must be opened.");
+
         this.partitionedFile = checkNotNull(partitionedFile);
         this.targetSubpartition = targetSubpartition;
+        this.dataFileChannel = checkNotNull(dataFileChannel);
+        this.indexFileChannel = checkNotNull(indexFileChannel);
 
         this.indexEntryBuf = ByteBuffer.allocateDirect(PartitionedFile.INDEX_ENTRY_SIZE);
         BufferReaderWriterUtil.configureByteBuffer(indexEntryBuf);
-
-        this.dataFileChannel = openFileChannel(partitionedFile.getDataFilePath());
-        try {
-            this.indexFileChannel = openFileChannel(partitionedFile.getIndexFilePath());
-        } catch (Throwable throwable) {
-            IOUtils.closeQuietly(dataFileChannel);
-            throw throwable;
-        }
     }
 
-    private FileChannel openFileChannel(Path path) throws IOException {
-        return FileChannel.open(path, StandardOpenOption.READ);
-    }
-
-    private boolean moveToNextReadableRegion() throws IOException {
-        if (currentRegionRemainingBuffers > 0) {
-            return true;
-        }
-
-        while (nextRegionToRead < partitionedFile.getNumRegions()) {
+    private void moveToNextReadableRegion() throws IOException {
+        while (currentRegionRemainingBuffers <= 0
+                && nextRegionToRead < partitionedFile.getNumRegions()) {
             partitionedFile.getIndexEntry(
                     indexFileChannel, indexEntryBuf, nextRegionToRead, targetSubpartition);
-            long dataOffset = indexEntryBuf.getLong();
+            nextOffsetToRead = indexEntryBuf.getLong();
             currentRegionRemainingBuffers = indexEntryBuf.getInt();
             ++nextRegionToRead;
-
-            if (currentRegionRemainingBuffers > 0) {
-                dataFileChannel.position(dataOffset);
-                return true;
-            }
         }
-
-        return false;
     }
 
     /**
-     * Reads a buffer from the {@link PartitionedFile} and moves the read position forward.
+     * Reads a buffer from the current region of the target {@link PartitionedFile} and moves the
+     * read position forward.
      *
      * <p>Note: The caller is responsible for recycling the target buffer if any exception occurs.
```

Comment:
Maybe add some comments to explain the params and returns for this public method?